core/ir/aarch64/codec.txt

```diff
@@ -294,1336 +294,1347 @@ x----------xxxxx----------------  wx16       # W/X register (or WZR/XZR)
+0111111011100000100010xxxxxxxxxx  n   62         cmge             d0 : d5
+01011110111xxxxx001111xxxxxxxxxx  n   62         cmge             d0 : d5 d16
+0x001110xx1xxxxx001101xxxxxxxxxx  n   63         cmgt            dq0 : dq5 dq16 bhsd_sz
+0x001110xx100000100010xxxxxxxxxx  n   63         cmgt            dq0 : dq5 bhsd_sz
+0101111011100000100010xxxxxxxxxx  n   63         cmgt             d0 : d5
+01011110111xxxxx001101xxxxxxxxxx  n   63         cmgt             d0 : d5 d16
+0x101110xx1xxxxx001101xxxxxxxxxx  n   64         cmhi            dq0 : dq5 dq16 bhsd_sz
+01111110111xxxxx001101xxxxxxxxxx  n   64         cmhi             d0 : d5 d16
+0x101110xx1xxxxx001111xxxxxxxxxx  n   65         cmhs            dq0 : dq5 dq16 bhsd_sz
+01111110111xxxxx001111xxxxxxxxxx  n   65         cmhs             d0 : d5 d16
+0111111011100000100110xxxxxxxxxx  n   66         cmle             d0 : d5
+0x101110xx100000100110xxxxxxxxxx  n   66         cmle            dq0 : dq5 bhsd_sz
+0x001110xx100000101010xxxxxxxxxx  n   67         cmlt            dq0 : dq5 bhsd_sz
+0101111011100000101010xxxxxxxxxx  n   67         cmlt             d0 : d5
+0x001110xx1xxxxx100011xxxxxxxxxx  n   68        cmtst            dq0 : dq5 dq16 bhsd_sz
+01011110111xxxxx100011xxxxxxxxxx  n   68        cmtst             d0 : d5 d16
+0x00111000100000010110xxxxxxxxxx  n   69          cnt            dq0 : dq5
+00011010110xxxxx010000xxxxxxxxxx  n   70       crc32b             w0 : w5 w16
+00011010110xxxxx010100xxxxxxxxxx  n   71      crc32cb             w0 : w5 w16
+00011010110xxxxx010101xxxxxxxxxx  n   72      crc32ch             w0 : w5 w16
+00011010110xxxxx010110xxxxxxxxxx  n   73      crc32cw             w0 : w5 w16
+10011010110xxxxx010111xxxxxxxxxx  n   74      crc32cx             w0 : w5 x16
+00011010110xxxxx010001xxxxxxxxxx  n   75       crc32h             w0 : w5 w16
+00011010110xxxxx010010xxxxxxxxxx  n   76       crc32w             w0 : w5 w16
+10011010110xxxxx010011xxxxxxxxxx  n   77       crc32x             w0 : w5 x16
+x0011010100xxxxxxxxx00xxxxxxxxxx  r   78         csel            wx0 : wx5 wx16 cond
+x0011010100xxxxxxxxx01xxxxxxxxxx  r   79        csinc            wx0 : wx5 wx16 cond
+x1011010100xxxxxxxxx00xxxxxxxxxx  r   80        csinv            wx0 : wx5 wx16 cond
+x1011010100xxxxxxxxx01xxxxxxxxxx  r   81        csneg            wx0 : wx5 wx16 cond
+110101010000100001111110010xxxxx  n   569     dc_cisw                : x0
+110101010000101101111110001xxxxx  n   570    dc_civac                : memx0
+110101010000100001111010010xxxxx  n   571      dc_csw                : x0
+110101010000101101111010001xxxxx  n   572     dc_cvac                : memx0
+110101010000101101111011001xxxxx  n   573     dc_cvau                : memx0
+110101010000100001110110010xxxxx  n   574      dc_isw                : x0
+110101010000100001110110001xxxxx  n   575     dc_ivac                : memx0
+110101010000101101110100001xxxxx  n   568      dc_zva          memx0 :
```

Comment:
Adding to Derek's comment in instr_create_api.h, I guess we need to use a new type of operand here instead of `memx0`. One that also back-aligns the address.

Reply:
Probably. I need to understand what I'm doing wrong as noted here (https://github.com/DynamoRIO/dynamorio/pull/5235#discussion_r761130002) before changing `memx0`.

Reply:
To be addressed and resolved in a post-v9.0 patch.